common/threadable.h

```diff
@@ -28,32 +29,39 @@ class Threadable
     //! Run on the current thread, blocking until process ends
     virtual void run() = 0;
 
+    //! Check if the run() function has been called
+    virtual bool isRunning()
+    {
+        return m_DoRun;
+    }
+
     //! Run the process on a background thread
     virtual void runInBackground()
     {
-        m_Thread = std::thread([this] { run(); });
-        m_ThreadRunning = true;
+        m_Thread = std::make_unique<std::thread>([this] { run(); });
```

Comment:
What is the purpose of moving ``m_Thread`` to the heap?

Reply:
It's not that I wanted ``m_Thread`` on the heap especially. I just made it a pointer so it could be ``nullptr`` and then we don't need a separate flag variable